### ISSUE TITLE
[BUG] No phone notificiations

### ISSUE THREAD

**Issue by reetp** [RocketChat/Rocket.Chat.Android#1111]
Your Rocket.Chat.Android version:
2.0.1
Rocket version 0.63.1 - Docker install

Phone:

Samsung S6 - Android 7.0

I have no phone notifications, and the Chat room notifications are inconsistent.

I fixed the 'No Messages yet' via https://github.com/RocketChat/Rocket.Chat.Android/issues/1100

Some Chats show a 'Number of unreads' and some don't. Doesn't seem consistent between Direct Chat and Groups either.

Without a numeric indicator there is not much difference in highlighting between a read and unread Chat (maybe an unread Chat in the list should be in a different colour?) - grey to black is not particularly distinct, at least on my phone.

I'll attach a screenshot shortly

**Comment by reetp** [comment]
![screenshot_20180418-111414-1](https://user-images.githubusercontent.com/3800384/38924049-3867c9cc-42fc-11e8-82f7-98730c53bc2d.jpg)

**Comment by reetp** [comment]
Hmmm....  Just seen this setting:

Always notify mobile
 True/False
Choose to always notify mobile device regardless of presence status.

Mine was set False which worked before. Changed to True and I now get phone notifications.

Maybe this should be set to default to True - it can then be disabled if it is found to be annoying....

**Comment by mmarif4u** [comment]
For the unread/read, there an issue. I guess it's in the works.
https://github.com/RocketChat/Rocket.Chat.Android/issues/864

**Comment by reetp** [comment]
Ah OK. I missed that !

**Comment by reetp** [comment]
I'm not sure of the logic between the General 'Always Notify Mobile' and then the individual 'Push Mobile Notifications for Mobile'.

Which has priority?

**Comment by antgel** [comment]
Are you using the back button to exit? Have you searched existing Issues such as https://github.com/RocketChat/Rocket.Chat.Android/issues/1078?

**Comment by reetp** [comment]
That may well be the issue. However, I just tried the following

Disable Always Notify
Push Mobile Notifications For All messages

Left screen on a chat (not going to Home)

User shows as online as auto away is false.

No sound notifications. No onscreen notifications.

Seems the only way I get sounds is with Disable Always Notify Enabled.

**Comment by rockneverdies55** [comment]
Just installed the native app and stopped receiving the push notifications. They used to work fine with the cordova app.

**Comment by rafaelks** [comment]
@rockneverdies55 @reetp Are you guys using our gateway to send notifications? Do you receive notifications if you **kill** the app?

**Comment by reetp** [comment]
Gateway - yes.

Kill the app - I get desktop notification

Start the app - I get desktop notification but no app notification

**Comment by rafaelks** [comment]
@reetp Kill the app, you get app notifications?

**Comment by rockneverdies55** [comment]
When I use the send test push notification to my user button in admin panel, it certainly is not working (whether app is killed or active).

But yesterday after I commented in here, I received couple of notifications from the app. So not sure what's going on. Can it be working sometimes and not always?

**Comment by rafaelks** [comment]
@robertwessen So here's the problem: the app is not receiving the **test notification** because it's not a valid chat notification, then the app ignores it. We'll have to "fix" that.

But looks like notifications are working for you.

**Comment by antgel** [comment]
@reetp What did you mean by "Left screen on a chat"? Until #1078 is fixed, I think only exiting the app with the back button will do.

**Comment by robertwessen** [comment]
@rafaelks think i got tagged on the wrong issue, did you mean @rockneverdies55 ?

**Comment by reetp** [comment]
"Left screen on a chat (not going to Home)"

Updated to 2.02

Trying to test a bit more it seems that if for any reason I show as online then notifications are not received, despite my profile setting saying:

"Push Mobile Notifications For" All Messages

I do not have auto away set, so if my desktop is left open I "remain" online and notifications do not get received on mobile

Equally unless the app is closed (#1078) you are left online which prevents notifications.

I guess the answer is really that the Profile setting should do what it says the same way that General/Notifications/Always notify mobile works.

"Push Mobile Notifications For" maybe should have extra settings eg:

Push all Mentions (when not online)
Push all Mentions regardless of status
Push all Messages (when not online)
Push all Messages regardless of status

**Comment by rafaelks** [comment]
This one is fixed in 2.1.0.

**Comment by reetp** [comment]
Can this be reopened as I am on 065.1 and Android 2.20 and notifications are still extremely hit and miss.

I'm trying to pin down exactly when and how, but not sure whether to do that here or open a new bug ?

**Comment by rafaelks** [comment]
@reetp We've completely rewritten the push notification logic in the client for 2.3.0 (currently in Beta), soon this version will hit the Google Play. You can start using it if you join our Beta program over there. 👍

**Comment by mmarif4u** [comment]
Hi @rafaelks, i know this issue is closed. Just to let know i am on 2.3.0 beta and noticed that when the app is closed, i don't get any notification. But if the app is open(in background) it receive notifications. Is this known to the dev team or should i open any issue?.

**Comment by rafaelks** [comment]
@mmarif4u What version of the server are you running? Also, are you using the gateway to send notifications?

**Comment by mmarif4u** [comment]
Rocket chat server : .65.1 snap  
Android ver : 2.3.0(on beta)

Yes, using RC gateway.

**Comment by rafaelks** [comment]
@mmarif4u This is strange. You never receive when the app is closed? Can you try with multiple messages?

**Comment by mmarif4u** [comment]
Yes, no push notification messages when app is closed.

I will try with multiple messages. But most of the messages in the mean time i get through email not push notification.

**Comment by reetp** [comment]
@rafaelks Thanks - I presume you mean Rocket.Chat (Experimental) (Unreleased) ? 
I'll give it a whirl :-)

Thanks

**Comment by rafaelks** [comment]
@reetp No, Experimental is another project (not related to this one). You need to go to this link (https://play.google.com/store/apps/details?id=chat.rocket.android) and opt-in for Beta in the bottom of the screen.

**Comment by sergiorykov** [comment]
Pushes. Again.

ENV: 
**server 0.66.3, ios app 3.0.1 (199), windows desktop client  2.11.0**

Server Settings: pushes go thru gateway, test push comes to my iphone app.
Personal Settings: pushes enabled for all types of messages

When desktop client is closed - all pushes go to my iphone app. **When desktop client is open - all notifications goes to desktop client and ios app doesn't receive any pushes....** 

It crashes all our efforts to use it as Slack replacement - I must be shure that my teammates will receive push in their phones.

**Comment by rafaelks** [comment]
@sergiorykov Just so I understand... the problem is that you don't receive pushes when the web/desktop client is connected? This is by design, if that's your problem.

**Comment by sergiorykov** [comment]
@rafaelks why? it's really important to receive pushes on mobile, even if desktop is open. I can forget to close desktop client and go home. I expect that push will find me at home with alert or question.

**Comment by rafaelks** [comment]
@sergiorykov I 100% agree with you. The web client should detect that there's no activity and change you to "away presence" but this seems to be failing in the latest releases... I see the same problem here. Can you please open an issue in our core repo about that? http://github.com/RocketChat/Rocket.Chat/

**Comment by sergiorykov** [comment]
OK)

**Comment by sergiorykov** [comment]
Oh, I've missed that I'm in Android repo))))

**Comment by sergiorykov** [comment]
Opened new issue https://github.com/RocketChat/Rocket.Chat/issues/11517

**Comment by antgel** [comment]
Perhaps connected to https://github.com/RocketChat/Rocket.Chat/issues/11378?

**Comment by rafaelks** [comment]
@antgel Good call! That may be it!

**Comment by stickboy06** [comment]
This may help people with Push notification problems, as I had to piece it together from multiple bug posts.
RocketChat updates have been changing settings that I have set, which then breaks functionality.  Push notifications for example stopped working after an update.  I searched for two months and updated to two different server versions during that time, but still they were broken.  One month I found that an update disabled and removed the Gateway server in the Push settings.  I enabled that and set it to https://gateway.rocket.chat, which then allowed my test Push notifications to work.  I still couldn't get push notifications to work naturally. After hours of testing, I found out the push notifications only worked when I closed out all other desktop/browser clients.  I then received push notifications that my according to my user account settings.  A month later, I found that the setting for Accounts > Default User Preferences > Enable Auto Away was set to False instead of True.  Push notifications are only sent when the user status is away or offline, so this setting broke notifications the way I had them setup before for our users.  I hope this helps some people "fix" their Push notification problems.
